include/net/mqtt_types.h

```diff
@@ -74,6 +74,9 @@ struct mqtt_connect_msg {
 	u16_t password_len;
 };
 
+/**
+ * @brief MQTT Publish Message
```

Comment:
A doxygen comment that (basically) just repeats the struct name isn't that useful

Reply:
I added those to fix reference issues, not that I was planning originally to fix them, the build fails with those missing, if you have a better suggestion, please let me know. :)